### PR TITLE
Metrics for MatrixHandler - Iline Kicks

### PR DESCRIPTION
**Pull request by Half-Shot**
To help us determine the damage when we kick lots of people for failing to get a client.

### CODE REVIEW

File: lib/bridge/IrcBridge.js [hunk 1]
```diff
@@ -181,11 +181,27 @@ IrcBridge.prototype._initialiseMetrics = function() {
...
                 this._clientPool.totalReconnectsWaiting(server.domain)
             );
+            let mxMetrics = this.matrixHandler.getMetrics(server.domain);
+            if (mxMetrics === null) {
+                mxMetrics = {}
+            }
+
+            matrixHandlerConnFailureKicks.set(
+                {server: server.domain},
+                mxMetrics["connection_failure_kicks"] ? mxMetrics["connection_failure_kicks"] : 0
```
Comment (turt2live) [inline lib/bridge/IrcBridge.js]: `mxMetrics["connection_failure_kicks"] || 0` also works

File: lib/bridge/MatrixHandler.js [hunk 2]
```diff
@@ -1481,6 +1485,22 @@ MatrixHandler.prototype._onUserQuery = Promise.coroutine(function*(req, userId)
     yield this.ircBridge.getMatrixUser(ircUser);
 });
 
+MatrixHandler.prototype._incrementMetric = function(serverDomain, metricName) {
+    let metricSet = this.metrics[serverDomain];
+    if (metricSet === undefined) {
+        metricSet = this.metrics[serverDomain] = {
+
```
Comment (turt2live) [inline lib/bridge/MatrixHandler.js]: the linter isn't actually forcing you to have this empty line, is it?

File: lib/bridge/MatrixHandler.js [hunk 3]
```diff
@@ -1481,6 +1485,22 @@ MatrixHandler.prototype._onUserQuery = Promise.coroutine(function*(req, userId)
     yield this.ircBridge.getMatrixUser(ircUser);
 });
 
+MatrixHandler.prototype._incrementMetric = function(serverDomain, metricName) {
+    let metricSet = this.metrics[serverDomain];
+    if (metricSet === undefined) {
```
Comment (turt2live) [inline lib/bridge/MatrixHandler.js]: actually can we just have this be a `!metricSet`?

File: lib/bridge/IrcBridge.js [hunk 4]
```diff
@@ -181,11 +181,27 @@ IrcBridge.prototype._initialiseMetrics = function() {
...
+    });
+
     metrics.addCollector(() => {
         this.ircServers.forEach((server) => {
             reconnQueue.set({server: server.domain},
                 this._clientPool.totalReconnectsWaiting(server.domain)
             );
+            let mxMetrics = this.matrixHandler.getMetrics(server.domain);
+            if (mxMetrics === null) {
+                mxMetrics = {}
```
Comment (turt2live) [inline lib/bridge/IrcBridge.js]: semicolon

File: lib/bridge/MatrixHandler.js [hunk 5]
```diff
@@ -1515,6 +1535,13 @@ MatrixHandler.prototype.onUserQuery = function(req, userId) {
     return reqHandler(req, this._onUserQuery(req, userId))
 };
 
+MatrixHandler.prototype.getMetrics = function(serverDomain) {
+    if (this.metrics[serverDomain] === undefined) {
```
Comment (turt2live) [inline lib/bridge/MatrixHandler.js]: This if statement isn't guarding anything

File: lib/bridge/IrcBridge.js (hunk 1 above)
Comment (Half-Shot) [reply lib/bridge/IrcBridge.js]: YES AAAAAARGH

**Pull request merged**